File: src/BaseHmiTypes/Screens/Controls/HmiRecipeControl.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiRecipeControl : HmiControlWindowBase
{
    public HmiRecipeControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiRecipeControl;
    }

    public HmiRecipeViewKind ViewKind { get; set; }

    public HmiProperty<bool>? ShowHeader { get; set; }

    public HmiProperty<bool>? ShowFooter { get; set; }

    public HmiProperty<int>? LinesPerItem { get; set; }

    public HmiProperty<bool>? WordWrap { get; set; }

    public HmiProperty<bool>? ViewOnly { get; set; }

    public HmiProperty<bool>? WrapAround { get; set; }

    public HmiProperty<HmiColor>? SelectionBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? SelectionForegroundColor { get; set; }

    public IList<HmiRecipeColumn> ColumnDefinitions { get; } = new List<HmiRecipeColumn>();
}

File: typescript/base-hmi-types/src/screens/controls/HmiRecipeControl.ts
import { HmiColor, HmiProperty } from "../base.js";
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiRecipeColumn } from "./HmiRecipeColumn.js";
import { HmiRecipeViewKind } from "./HmiRecipeViewKind.js";

export class HmiRecipeControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiRecipeControl;
  }

  viewKind = HmiRecipeViewKind.Selector;
  showHeader?: HmiProperty<boolean>;
  showFooter?: HmiProperty<boolean>;
  linesPerItem?: HmiProperty<number>;
  wordWrap?: HmiProperty<boolean>;
  viewOnly?: HmiProperty<boolean>;
  wrapAround?: HmiProperty<boolean>;
  selectionBackgroundColor?: HmiProperty<HmiColor>;
  selectionForegroundColor?: HmiProperty<HmiColor>;
  readonly columnDefinitions: HmiRecipeColumn[] = [];
}
